packages/grafana-e2e/src/support/types.ts

```diff
@@ -55,28 +59,31 @@ const processSelectors = <S extends Selectors>(e2eObjects: E2EFunctions<S>, sele
 
     if (typeof value === 'string') {
       // @ts-ignore
-      e2eObjects[key] = () => {
+      e2eObjects[key] = (options?: CypressOptions) => {
         logOutput(value);
-        return e2e().get(Selector.fromAriaLabel(value));
+        return e2e().get(Selector.fromAriaLabel(value), options);
       };
 
       continue;
     }
 
     if (typeof value === 'function') {
       // @ts-ignore
-      e2eObjects[key] = (text?: string) => {
-        if (!text) {
-          const selector = value((undefined as unknown) as string);
-
-          logOutput(selector);
-          return e2e().get(selector);
+      e2eObjects[key] = (textOrOptions?: string | CypressOptions, options?: CypressOptions) => {
+        // set the selector to default if we are seeing () => or (options) =>
+        let selector = value((undefined as unknown) as string);
+
+        // if (text) => or (text, options) => then we want to update the selector
+        if (typeof textOrOptions === 'string') {
+          const ariaText = value(textOrOptions);
+          selector = Selector.fromAriaLabel(ariaText);
+        } else {
+          // textOrOptions is the options passed in so overwrite the options param
+          options = textOrOptions;
```

Comment:
I wonder if this would become less complex if we checked for arguments length? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/length

Reply:
I had a think with the `arguments.length` check and maybe I'm overcomplicating things with it? 😄 If we checked for the arguments length, wouldn't we still need to check for the type of the first param as well?

My understanding of the behaviour we want would be something like this (but tidied up 🙃 ):
```
        // the input can only be ()
        if (arguments.length === 0) {
          const selector = value((undefined as unknown) as string);
          return e2e().get(selector);
        }

        // the input can be (text) or (options)
        if (arguments.length === 1) {
          if (typeof textOrOptions === 'string') {
            const ariaText = value(textOrOptions);
            const selector = Selector.fromAriaLabel(ariaText);
            return e2e().get(selector);
          }
          const selector = value((undefined as unknown) as string);
          return e2e().get(selector, textOrOptions);
        }

        // the input can only be (text, options)
        if (arguments.length === 2) {
          // need to cast textOrOptions or check it's a string here or ts will complain
          const ariaText = value(textOrOptions as string);
          const selector = Selector.fromAriaLabel(ariaText);
          return e2e().get(selector, options);
        }
```

Am I missing something? Is there a way this could be refactored so it's not as complicated as the one `if (typeof textOrOptions === 'string')` check here?

Reply:
I think is just a bit more complex, but a lot more readable for someone reading it for the first time, so I vote for using the arguments length. 
But maybe this brings up how bad the design of e2e wrapper is 😄  so in the long run maybe we should look into improving that.

Reply:
Makes sense - I agree, it's longer but more readable so I'll update this 👍